compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -23,10 +23,11 @@ use rustc_trait_selection::traits::query::evaluate_obligation::InferCtxtExt as _
 
 impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     pub(crate) fn body_fn_sig(&self) -> Option<ty::FnSig<'tcx>> {
+        let body_hir = self.tcx.hir().local_def_id_to_hir_id(self.body_id);
         self.typeck_results
             .borrow()
             .liberated_fn_sigs()
-            .get(self.tcx.hir().get_parent_node(self.body_id))
+            .get(self.tcx.hir().get_parent_node(body_hir))
```

Comment:
```suggestion
            .get(body_hir)
```
New `self.body_id` is already one level higher than the former one.